src/lib/pages/frontsite/PatronProfile/PatronCurrentLoans/BrwReqLoanExtendButton.js

```diff
@@ -0,0 +1,215 @@
+    const extensionStatus = extension.status;
+    const isPending = extensionStatus === 'PENDING';
+    if (isPending) {
+      isValid = false;
+      const requestDate = extension.request_date;
+      msg = REQUEST_MESSAGES.IS_PENDING(requestDate);
+    }
+
+    const isDeclined = extensionStatus === 'DECLINED';
+    if (isDeclined) {
+      isValid = false;
+      msg = REQUEST_MESSAGES.IS_DECLINED;
+    }
+  }
+
+  return {
+    isValid: isValid,
+    cmp: msg ? (
+      <>
+        <span>{msg}</span>{' '}
+      </>
+    ) : null,
+  };
+};
+
+class BrwReqLoanExtendButton extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      requestExtensionIsLoading: false,
+      brwReqDetailsIsLoading: false,
+      brwReqDetails: {},
+      brwReqDetailsError: {},
+    };
+    this.borrowingRequestFetcher = new BorrowingRequestFetcher();
+    this.extensionRequester = new ExtensionRequester();
+  }
+
+  componentDidMount() {
+    const { brwReqLoan } = this.props;
+    const brwReqPid = brwReqLoan.metadata.item_pid.value;
+    this.fetchBorrowingRequestDetails(brwReqPid);
+  }
+
+  componentWillUnmount() {
+    this.borrowingRequestFetcher.cancel();
+    this.extensionRequester.cancel();
+  }
+
+  fetchBorrowingRequestDetails = async brwReqPid => {
+    this.setState({ brwReqDetailsIsLoading: true });
+    try {
+      const data = await this.borrowingRequestFetcher.get(brwReqPid);
+      this.setState({ brwReqDetails: data });
+    } catch (error) {
+      this.setState({ brwReqDetailsError: error });
+    } finally {
+      this.setState({ brwReqDetailsIsLoading: false });
+    }
+  };
+
+  triggerExtensionRequest = async () => {
+    const { brwReqLoan, onSuccess, onError } = this.props;
+    const brwReqPid = brwReqLoan.metadata.item_pid.value;
+    this.setState({ requestExtensionIsLoading: true });
+    try {
+      await this.extensionRequester.doRequest(brwReqPid);
+      const documentTitle = brwReqLoan.metadata.document.title;
+      onSuccess(INFO_MESSAGES.SUCCESS(documentTitle));
+    } catch (error) {
+      onError(error.response.data.message);
+    } finally {
+      this.setState({ requestExtensionIsLoading: false });
+    }
+  };
+
+  render() {
+    const { brwReqLoan } = this.props;
+    const {
+      requestExtensionIsLoading,
+      brwReqDetailsIsLoading,
+      brwReqDetails,
+      brwReqDetailsError,
+    } = this.state;
+
+    const isLoading = brwReqDetailsIsLoading || requestExtensionIsLoading;
+
+    const loanExtensionValidation = validateLoanExtension(brwReqLoan);
+
+    const extension = _get(brwReqDetails, 'metadata.patron_loan.extension', {});
+    const extensionRequestValidation = validateExtensionRequest(extension);
+
+    const isValid =
+      loanExtensionValidation.isValid && extensionRequestValidation.isValid;
+    const isDisabled = !isValid;
+
+    return (
+      <Error error={brwReqDetailsError}>
+        {!isLoading && extensionRequestValidation.cmp}
```

Comment:
it bugs me a bit that I don't know immediately what is the `.cmp` displayed, but I don't have a better idea for now